### PR TITLE
[DNM]Overhauls death and dying completely.

### PR DESCRIPTION
**Pull request by Iamgoofball**
- Completely overhauls death and dying to be more fun for the person dying.
- When you go into crit, you start taking 2 oxygen damage a tick, start stuttering, and roll for 4 potential things to happen. The severity and chances of these effects go up the longer you're in crit.
  - Knocks you out for X amount of seconds. Maximum is 2. Minimum is 0.
  - Gives you X amount of losebreath and Y amount of brain damage. Max losebreath is 8 and max brain damage is 2. Minimum losebreath is 1 and minimum brain damage is 0.
  - X percent chance to have a heart attack. If you are already having a heart attack, you get Y losebreath instead. Maximum heart attack chance is 45%, maximum losebreath is 8. Minimum heart attack is 0%, minimum losebreath is 1.
  - Another roll for "Knocks you out for X amount of seconds." Maximum is 2. Minimum is 0.
- If you have -100 health, it starts rolling a chance for you to die. Your chance to die goes up the more brain loss you have.
- When you get out of crit, you start healing oxygen loss and losing losebreath. This allows you to recover from critical. When you finish healing those you stop recovering oxygen loss and losebreath.
  The benefit of all this is that when you enter crit, you can attempt to crawl to medbay or attempt to get a few last punches out on your attacker.

Only a masochist would enjoy staring at a black screen for 10 minutes, but considering this is me coding, I'm certain you guys are gonna claim that staring at blank screens for 10 minutes is enjoyable gameplay for everyone. :^)

This is a pretty big change that might be a bit disliked:
- Uncaps the amount of damage you can take in every damage type. You can have 500 brute damage now no problem. Before it was capped at 100.
- Brain Loss is now lethal.
- At 100 brain loss, you get weakened every tick and start taking losebreath and cannot speak.
- At 120 brain loss, you die of brain failure.
- Previously, the only thing brain loss did was "lol gibbering joke eecks dee xD". A lot of things used this as a buffer or gimmick damage type. Please tell me locations of brain damage's use so I can remove it or nerf it into reason now that brain damage is an actually worrying thing like all the other damages except cloneloss
  (remove cloneloss when)
- You still do the gibbering at 60 brain loss or higher. I'm not that mean. Yet.

Minor changes:
- Heart attacks now do 0.5 brain loss instead of 1 brute loss since brain loss is now lethal.

Q&A:
- How is this fun for the medical doctor?

Your patients are now more durable and won't die on you immediately after they hit a dedicated "health death threshold".
Your patients in critical condition can try to pull themselves to the medbay for quicker treatment.
- How is this fun for the dying patient?

You aren't staring at a blank screen for 10 minutes watching white creep up on your screen, or immediately hitting "succumb" when you go into crit, because you can now attempt to crawl to safety or get that last hit out on your attacker.
You are engaged in the game a bit more when dying because you're still actively watching the game because there's interesting things going on, as you can still stumble around and crawl places.

### CODE REVIEW

**Comment by MrPerson** [comment]
Seems way too complicated for something that could have just been a simple "You take breathloss damage while in crit" instead of this "You roll probabilities to take brain damage (?) and eventually die after a randomized amount of time" nonsense.

**Comment by WJohn** [comment]
To what extent does brainrot deal brain damage? What about brain-damaging chems, or cloning?

**Comment by Cheridan** [comment]
>Brain Loss is now lethal.

Fuck no. This was removed because a brainrot infection would debilitate and kill the entire station. Brainloss is SUPPOSED to be a gimmick.

>Uncaps the amount of damage you can take in every damage type.

Are limb damage caps still in place? I think it's good/wealwistic that you can't kill someone by hitting them in the leg over and over again.

Otherwise this seems ok, should be a lot more cinematic than *plop*

**Comment by Midaychi** [comment]
I like it, and you can't powergame it really. It's actually not as complicated as you make it sound in the description. 

The change goes from "pow you're on the ground blacked out" to "Oh god why crawl to medbay" or "HALP... GRIFFIN *disarm disarm* ..WHY ..COM NO WORK"

Seems like it might make changling/antags actually have to work for their kills though instead of using "lol win" buttons. 

Problem though! What if the person getting the shit kicked out of them has handy medical supplies on hand? All they need is a few seconds and they can get themselves out of crit right? This is babymode tg medical we're talking about.

Although
"If you have -100 health, it starts rolling a chance for you to die. This is your brain loss divided by 4."

So, having more brain damage means you're less likely to die?

**Comment by Kromgar** [comment]
Okay well nerf brainrot to only rot the brain to a certain point then Cheridan

**Comment by PKPenguin321** [comment]
death: RNG edition
it's not a *bad* system but imo not as good as ours which is more straightforward

**Comment by Iamgoofball** [comment]
@Midaychi no, your chance to die goes up with the more brain loss you have

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -244,14 +244,18 @@ Sorry Giacom. Please don't be mad :(
 
 /mob/living/proc/adjustBruteLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	bruteloss = min(max(bruteloss + amount, 0),(maxHealth*2))
+	bruteloss += amount
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: You've removed the Clamp() (what was there before is equivalent to Clamp()), when you really only needed to remove the cap.

change it to bruteloss = max(0, bruteloss+amount), instead of checking if it's below 0 seperately

**Comment by ACCount12** [comment]
Too much random. And it's really easy to survive your crit.

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -244,14 +244,18 @@ Sorry Giacom. Please don't be mad :(
...
+		bruteloss = 0
 
 /mob/living/proc/getOxyLoss()
 	return oxyloss
 
 /mob/living/proc/adjustOxyLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	oxyloss = min(max(oxyloss + amount, 0),(maxHealth*2))
+	oxyloss += amount
+	if(oxyloss <= 0)
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -262,7 +266,9 @@ Sorry Giacom. Please don't be mad :(
 
 /mob/living/proc/adjustToxLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	toxloss = min(max(toxloss + amount, 0),(maxHealth*2))
+	toxloss += amount
+	if(toxloss <= 0)
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -273,14 +279,18 @@ Sorry Giacom. Please don't be mad :(
 
 /mob/living/proc/adjustFireLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	fireloss = min(max(fireloss + amount, 0),(maxHealth*2))
+	fireloss += amount
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above

File: code/modules/mob/living/living.dm [hunk 5]
```diff
@@ -273,14 +279,18 @@ Sorry Giacom. Please don't be mad :(
...
+	if(fireloss <= 0)
+		fireloss = 0
 
 /mob/living/proc/getCloneLoss()
 	return cloneloss
 
 /mob/living/proc/adjustCloneLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	cloneloss = min(max(cloneloss + amount, 0),(maxHealth*2))
+	cloneloss += amount
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above

File: code/modules/mob/living/living.dm [hunk 6]
```diff
@@ -291,7 +301,9 @@ Sorry Giacom. Please don't be mad :(
 
 /mob/living/proc/adjustBrainLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	brainloss = min(max(brainloss + amount, 0),(maxHealth*2))
+	brainloss  += amount
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above, Yaawn.

File: code/modules/mob/living/living.dm [hunk 7]
```diff
@@ -302,7 +314,9 @@ Sorry Giacom. Please don't be mad :(
 
 /mob/living/proc/adjustStaminaLoss(var/amount)
 	if(status_flags & GODMODE)	return 0
-	staminaloss = min(max(staminaloss + amount, 0),(maxHealth*2))
+	staminaloss += amount
```
Comment (RemieRichards) [inline code/modules/mob/living/living.dm]: Above

**Comment by Incoming5643** [comment]
It seems really silly that a guy could have thousands of brute damage but still be alive because he hadn't been mutilated long enough for brain damage to set in. Other than that it is interesting at least, but would require so much playtesting and a rebalancing of SO much shit in relation to it.

**Comment by Midaychi** [comment]
Yeah, this definitely needs to be tested on a test server with a few rounds of nuke and changeling first to see how it works in practice.

**Comment by SconesC** [comment]
Honestly crit should just be unconsciousness as-is. You take oxy damage, and nowadays typically blood loss, and it seems fine to me. 

It's going to be a pain in the ass to kill people now because of this, and I don't think that's a good thing. It's just another few seconds of dumping another 100 damage into them until they die. This needs intensive testing with consideration to nuke ops especially seeing as they literally run out of bullets. And that is not a good thing. I should not NOT be able to kill the Captain because I ran out of bullets and he gets good RNG with his psuedo-crit while I drag him around beating him

...That, or ops need standard-issue melee weapons.

**Comment by Iamgoofball** [comment]
>It's just another few seconds of dumping another 100 damage into them until they die.
uhm

that's what we have right now, dude

right now on the live server you enter crit in 0 hp, and another 100 damage kills you

testing is definitely planned, getting the server owners on board with it is required first though

**Comment by RemieRichards** [comment]
@Iamgoofball actually no, since the "another 100 damage" for CURRENT Crit, the guy is on the floor and has 0 way of retaliating.
With this new method he can retaliate until he actually dies (which could be after you've put 0 extra damage in or 50 or whatever RNG)

**Comment by Iamgoofball** [comment]
yes, I meant current crit

**Comment by dannno** [comment]
i hope this works out because current crit sucks

**Comment by RemieRichards** [comment]
So what we've got on the server now is a Technical death at 100 HP, with a real death at 200HP.
Yours removes technical death, which could actually be insane if the RNG doesn't trigger.

**Comment by WJohn** [comment]
I for one would prefer if you could only crawl while in this "technical death" and could not get up to pick up items. You shouldn't be able to fight at this point, so combat is hardly changed. On the other hand, if you're poisoned or are getting away from a hull breach you might last some time longer and find help.

File: code/modules/mob/living/carbon/life.dm [hunk 8]
```diff
@@ -502,3 +504,114 @@
...
+				world << "stage10"
+			if(-INFINITY to -95)
+				paralyse_amount = 2
+				losebreath_amount = 8
+				cardiac_arrest_chance = 45
+				brain_damage_from_crit = 2
+				world << "stage11"
+		var/picked_effect = rand(1,4)
+		switch(picked_effect)
+			if(1)
```
Comment (MrPerson) [inline code/modules/mob/living/carbon/life.dm]: Just make this if(1,4) or better yet 1,2

File: code/modules/mob/living/carbon/life.dm [hunk 9]
```diff
@@ -502,3 +504,114 @@
...
+				Paralyse(paralyse_amount)
+		if(!reagents.has_reagent("epinephrine"))
+			adjustBrainLoss(brain_damage_from_crit)
+			world << "dealing [brain_damage_from_crit] of brain loss"
+		stuttering += 5
+		if(!reagents.has_reagent("epinephrine"))
+			adjustOxyLoss(2)
+		var/deathprob = (getBrainLoss()/4)
+		if(prob(deathprob) && health <= -100)
+			world << "CRITDEATH [deathprob]"
```
Comment (MrPerson) [inline code/modules/mob/living/carbon/life.dm]: lel

File: code/modules/mob/living/carbon/life.dm [hunk 10]
```diff
@@ -502,3 +504,114 @@
...
+						heart_attack = 1
+					else
+						losebreath += losebreath_amount
+			if(4)
+				Paralyse(paralyse_amount)
+		if(!reagents.has_reagent("epinephrine"))
+			adjustBrainLoss(brain_damage_from_crit)
+			world << "dealing [brain_damage_from_crit] of brain loss"
+		stuttering += 5
+		if(!reagents.has_reagent("epinephrine"))
```
Comment (MrPerson) [inline code/modules/mob/living/carbon/life.dm]: Do these two checks at the same time so you don't have to test for epinephrine twice

File: code/modules/mob/living/carbon/life.dm (hunk 9 above)
Comment (Incoming5643) [inline code/modules/mob/living/carbon/life.dm]: This is needless hair splitting and probably not even a thing that works, but I always put prob()'s last in && chains so the game doesn't roll RNG unless it absolutely needs to.

File: code/modules/mob/living/carbon/life.dm [hunk 11]
```diff
@@ -502,3 +504,114 @@
...
+					if(!heart_attack)
+						src << "<span class = 'userdanger'>Your heart stops!</span>"
+						heart_attack = 1
+					else
+						losebreath += losebreath_amount
+			if(4)
+				Paralyse(paralyse_amount)
+		if(!reagents.has_reagent("epinephrine"))
+			adjustBrainLoss(brain_damage_from_crit)
+			world << "dealing [brain_damage_from_crit] of brain loss"
```
Comment (Incoming5643) [inline code/modules/mob/living/carbon/life.dm]: Doesn't this mean that a mob chock full of epinephrine is utterly immune to death from damage?

**Comment by Kromgar** [comment]
>Crawling away from a hullbreach
>Implying space wind isnt op

File: code/modules/mob/living/carbon/life.dm (hunk 9 above)
Comment (MrPerson) [inline code/modules/mob/living/carbon/life.dm]: It actually does work and can be useful in large, expensive loops or things like that. Doesn't really matter one way or the other here though because it's minor in general.

File: code/modules/mob/living/carbon/life.dm [hunk 12]
```diff
@@ -270,14 +270,16 @@
...
-
-		if(getOxyLoss() > 50 || health <= config.health_threshold_crit)
-			Paralyse(3)
-			stat = UNCONSCIOUS
-
+		if(getBrainLoss() >= 100 && stat != DEAD)
+			Weaken(20)
+			losebreath += 10
+			silent += 2
+		if(getBrainLoss() >= 120 && stat != DEAD)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/life.dm]: Will have to make adjustments to the cloner if you go with this, otherwise cloned people will immediately drop dead from starting off with 190 brain damage.

**Comment by WJohn** [comment]
You get the idea, it still improves your chances to SOME extent.

**Comment by Midaychi** [comment]
Making crit force crawl would make it so you'd have to buckle yourself to something in order to handle objects, and if you're in crit you'd be gasping and find it hard to comms, so that might be the fixer to not immediate unconsciousness.

File: code/modules/mob/living/carbon/life.dm (hunk 9 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/life.dm]: debug thingy was having problems with shit not getting called

File: code/modules/mob/living/carbon/life.dm (hunk 10 above)
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/life.dm]: k

**Comment by Lumipharon** [comment]
It's bad because you can just carry medical supplies and never worry about dying, unless they do hundreds of damage + to you.

Also why should dying be fun?

Current crit is you're unconscious and not breathing - so you suffocate to death.
Gooncrit is you take assloads of damage and keep swinging/callin for help/doing shit, until your brain melts out your eyeballs.

Literally everything is balanced around our current system, please don't change it.

**Comment by Steelpoint** [comment]
Gooncrit has it that you can't use any items while in crit though.

**Comment by Fox-McCloud** [comment]
@Lumipharon 

Actually, on Goon's system, if you have more than X about of O2 damage you can't talk--and since you lose consciousness frequently in their crit system, it doesn't take long, at all, to not be able to talk--you may be able to fight back so some extent, but in general, it wont' be viable for you to do so considering how much you'll be falling over.

it generally forces people to commit more to kill someone rather than just "shoot 5 times, move to next one, shoot 5 times, move to next one".

**Comment by AnturK** [comment]
* Makes aliens literally immortal
* Breaks ling revive
* Makes defibs/any other resuscitation useless or take 5x times longer
* You can perform CPR on yourself to stay alive pretty much indefinitely

This is pretty pointless change, in murder scenario it's only annoying to the killer, in accidents this is at least doubling of health.

**Comment by Jalleo** [comment]
Goof have this work woth the level of blood people have because it is a key part if you want this to work well.
Not to mention healing brain damage is basically non existant.
It really should be based on multiple damages so you dont have bad rng

**Comment by Steelpoint** [comment]
Brain damage is not too hard to fix, a drop of Mannitol in the Cyro Tube/Pill Bottle should work.

**Comment by bgobandit** [comment]
The chaplain is now the most deadly member of the crew.

**Comment by Steelpoint** [comment]
Mannitol is really easy to make, so brain damage is not a massive problem.

As I said on the forums, I feel a crit system change needs a good design document but also a very extensive testing period, and I don't think either of those have been satisfied. I agree that our crit system can do with a change however.

**Comment by Fox-McCloud** [comment]
bgobandit---except the part where he's going to have to beat someone, on average, 25 times to knock them out and 30 times to out and out kill them....when his null rod will knock them out in 7 and kill in 14.

**Comment by PKPenguin321** [comment]
@Fox-McCloud but considering nobody else on the crew has the means to reliably kill anyone anymore because everyone's immortal now, chaplain's the only one that can kill at all, meaning he's the most deadly :^)

**Comment by Fox-McCloud** [comment]
@PKPenguin321 

Humph! =p

**Comment by Iamgoofball** [comment]
I'm gonna close this.
